### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by serebryakov02**
lab1 finished

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+x64/
+.vs/
+*.sln
+*.vcxproj
+*.vcxproj.filters
+*.vcxproj.user
```
Comment (ArtemLubimov) [inline .gitignore:6]: New line in the end of file

File: lab1/Factory Method/Employee.h [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#pragma once
+#ifndef EMPLOYEE_H_
+#define EMPLOYEE_H
+
+#include <iostream>
+
```
Comment (ArtemLubimov) [inline lab1/Factory Method/Employee.h:6]: Add comments what it is

File: lab1/Factory Method/Format.cpp [hunk 3]
```diff
@@ -0,0 +1,37 @@
+#include "Format.h"
+#include <iostream>
+
+void TextFormat::display(const Employee& e) const
+{
+	std::cout << "Text Format!\n" << e;
+}
+
+void XMLFormat::display(const Employee& e) const
```
Comment (ArtemLubimov) [inline lab1/Factory Method/Format.cpp:9]: Add comments

File: lab1/Factory Method/Format.h [hunk 4]
```diff
@@ -0,0 +1,33 @@
+#pragma once
+#ifndef FORMAT_H_
+#define FORMAT_H_
+
+#include "Employee.h"
+
```
Comment (ArtemLubimov) [inline lab1/Factory Method/Format.h:6]: Add comments what is about

File: lab1/Factory Method/Workshop.h [hunk 5]
```diff
@@ -0,0 +1,32 @@
+#pragma once
+#ifndef WORKSHOP_H_
+#define WORKSHOP_H_
+
+#include "Format.h"
+
+class Production
```
Comment (ArtemLubimov) [inline lab1/Factory Method/Workshop.h:7]: add comments what is here

File: lab1/Factory Method/main.cpp [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+{
+	std::vector<Employee> employees;
+	employees.push_back(Employee(1, "Senior C++ Developer",
+		"Nikita Serebryakov"));
+	employees.push_back(Employee(2, "Director", "Larisa Izotova"));
+
+	for (const auto& a : employees)
+		std::cout << a << '\n';
+
+	Production* creator = new TextWorkshop();
```
Comment (ArtemLubimov) [inline lab1/Factory Method/main.cpp:16]: Add comment what is here

File: lab1/Factory Method/main.cpp [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+	Format* txtFmt = creator->create();
+	delete creator;
+	creator = new XMLWorkshop();
+	Format* xmlFmt = creator->create();
+	delete creator;
+	creator = new HTMLWorkshop();
+	Format* htmlFmt = creator->create();
+	delete creator;
+
+	/*txtFmt->display();
```
Comment (ArtemLubimov) [inline lab1/Factory Method/main.cpp:26]: IF its commented - it means it's not needed - why is it still here?

File: lab1/Factory Method/main.cpp [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+		}
+		std::cout << '\n';
+	}
+
+	delete txtFmt;
+	delete xmlFmt;
+	delete htmlFmt;
+
+	return 0;
+}
```
Comment (ArtemLubimov) [inline lab1/Factory Method/main.cpp:62]: Non new line in the end

File: lab1/Prototype/FootballPlayers.cpp [hunk 9]
```diff
@@ -0,0 +1,157 @@
+#include "FootballPlayers.h"
+
+// Methods of the FootballPlayer class
+
+FootballPlayer::FootballPlayer(const FootballPlayer& f)
```
Comment (ArtemLubimov) [inline lab1/Prototype/FootballPlayers.cpp:5]: Each player type - separate class

File: lab1/Prototype/FootballPlayers.h [hunk 10]
```diff
@@ -0,0 +1,72 @@
+#pragma once
+#ifndef FOOTBALLPLAYER_H_
+#define FOOTBALLPLAYER_H_
+
+#include <iostream>
+#include <string>
+
+class FootballPlayer
```
Comment (ArtemLubimov) [inline lab1/Prototype/FootballPlayers.h:8]: The same here - separate classes and comments

File: lab1/Prototype/main.cpp [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+#include <vector>
+
+const int SIZE = 3;
+
+int main()
+{
+	using std::cout;
+	using std::cin;
+
+	/*Goalkeeper* gDonor = new Goalkeeper();
```
Comment (ArtemLubimov) [inline lab1/Prototype/main.cpp:12]: If it's commented - it's not needed

File: lab1/Prototype/main.cpp [hunk 12]
```diff
@@ -0,0 +1,94 @@
+#include "FootballPlayers.h"
```
Comment (ArtemLubimov) [inline lab1/Prototype/main.cpp:1]: Order of imports - in the beginning native libraries. after that headers your own

File: lab1/Prototype/main.cpp [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+		"2 - Defender\n"
+		"3 - Midfielder\n"
+		"4 - Attack\n"
+		"*Any characher to quit*\n";
+	int choice, counter = 0;
+	while (counter < SIZE && cin >> choice && (choice >= 1 && choice <= 4))
+	{
+		system("cls");
+		cin.get();
+
```
Comment (ArtemLubimov) [inline lab1/Prototype/main.cpp:40]: add comments

File: lab1/Prototype/main.cpp [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+			"*Any characher to quit*\n";
+	}
+
+	cout << "\nYour team:\n";
+	for (auto it = team.begin(); it != team.end(); it++)
+		cout << "Full name: " << (*it)->getFullName() <<
+		"\nPosition: " << (*it)->getPosition() << '\n';
+
+	return 0;
+}
```
Comment (ArtemLubimov) [inline lab1/Prototype/main.cpp:94]: No new line in the end

File: .gitignore (hunk 1 above)
Comment (serebryakov02) [reply .gitignore:6]: Done.

File: lab1/Prototype/main.cpp (hunk 13 above)
Comment (serebryakov02) [reply lab1/Prototype/main.cpp:31]: Done.

File: lab1/Prototype/main.cpp (hunk 14 above)
Comment (serebryakov02) [reply lab1/Prototype/main.cpp]: Done.

File: lab1/Factory Method/Employee.h (hunk 2 above)
Comment (serebryakov02) [reply lab1/Factory Method/Employee.h:6]: Done.

File: lab1/Factory Method/Format.cpp (hunk 3 above)
Comment (serebryakov02) [reply lab1/Factory Method/Format.cpp:9]: Done.

File: lab1/Factory Method/Format.h (hunk 4 above)
Comment (serebryakov02) [reply lab1/Factory Method/Format.h:6]: Done.

File: lab1/Prototype/main.cpp (hunk 11 above)
Comment (serebryakov02) [reply lab1/Prototype/main.cpp]: Deleted.

File: lab1/Factory Method/Workshop.h (hunk 5 above)
Comment (serebryakov02) [reply lab1/Factory Method/Workshop.h:9]: Done.

File: lab1/Factory Method/main.cpp (hunk 7 above)
Comment (serebryakov02) [reply lab1/Factory Method/main.cpp]: Edited.

**Review by ArtemLubimov — APPROVED** [review]

File: lab2/Bridge/Video.cpp [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+{
+}
+
+Video::Video(const std::string& name, const std::string& format,
+	double duration, double size)
+	: name_(name), format_(format), duration_(duration), size_(size)
+{
+}
+
+//Video::Video(const Video& video)
```
Comment (ArtemLubimov) [inline lab2/Bridge/Video.cpp:14]: commented - remove it

File: lab2/Bridge/Video.h [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+private:
+	std::string name_;
+	std::string format_;
+	double duration_;	// In seconds.
+	double size_;		// In megabytes.
+public:
+	Video();
+	Video(const std::string& name, const std::string& format,
+		double duration, double size);
+	//Video(const Video& video);
```
Comment (ArtemLubimov) [inline lab2/Bridge/Video.h:17]: commented? you know what to do

File: lab2/Bridge/Video.h [hunk 17]
```diff
@@ -0,0 +1,26 @@
+#ifndef VIDEO_H_
+#define VIDEO_H_
+
```
Comment (ArtemLubimov) [inline lab2/Bridge/Video.h:3]: where is comments?

File: lab2/Bridge/main.cpp [hunk 18]
```diff
@@ -0,0 +1,24 @@
+#include "Application.h"
+
+int main()
+{
+	StaticThema* lightTema = new StaticThema;
```
Comment (ArtemLubimov) [inline lab2/Bridge/main.cpp:5]: Where is comments?

File: lab2/Bridge/Video.h [hunk 19]
```diff
@@ -0,0 +1,26 @@
...
+	//Video(const Video& video);
+	void setVideo(const std::string& name, const std::string& format,
+		double duration, double size);
+	std::string getName() const;
+	std::string getFormat() const;
+	double getDuration() const;
+	double getSize() const;
+};
+
+#endif // !VIDEO_H_
```
Comment (ArtemLubimov) [inline lab2/Bridge/Video.h:26]: No newline in end of file

File: lab2/Bridge/main.cpp [hunk 20]
```diff
@@ -0,0 +1,24 @@
...
+	app.displayApp();
+	app.setThema(darkThema);
+	std::cout << "2:\n";
+	app.displayApp();
+
+	delete lightTema;
+	delete darkThema;
+
+	return 0;
+}
```
Comment (ArtemLubimov) [inline lab2/Bridge/main.cpp:24]: No newline in end of file

File: lab2/Proxy/Site.cpp [hunk 21]
```diff
@@ -0,0 +1,37 @@
+#include "Site.h"
+
```
Comment (ArtemLubimov) [inline lab2/Proxy/Site.cpp:2]: Comments what this file about

File: lab2/Proxy/Site.h [hunk 22]
```diff
@@ -0,0 +1,33 @@
+#ifndef SITE_H_
+#define SITE_H_
+
```
Comment (ArtemLubimov) [inline lab2/Proxy/Site.h:3]: Comments what this file about

File: lab2/Bridge/Video.cpp (hunk 15 above)
Comment (serebryakov02) [reply lab2/Bridge/Video.cpp:14]: Done.

File: lab2/Bridge/Video.h (hunk 16 above)
Comment (serebryakov02) [reply lab2/Bridge/Video.h:17]: Yes, it needs to remove, done.

File: lab2/Bridge/Video.h (hunk 17 above)
Comment (serebryakov02) [reply lab2/Bridge/Video.h:3]: Added.

File: lab2/Bridge/main.cpp (hunk 18 above)
Comment (serebryakov02) [reply lab2/Bridge/main.cpp:5]: Added a short comment.

File: lab2/Proxy/Site.h (hunk 22 above)
Comment (serebryakov02) [reply lab2/Proxy/Site.h:3]: Done.

File: lab2/Bridge/Video.h (hunk 19 above)
Comment (serebryakov02) [reply lab2/Bridge/Video.h:26]: Added.

File: lab2/Bridge/main.cpp (hunk 20 above)
Comment (serebryakov02) [reply lab2/Bridge/main.cpp:24]: Added.

**Pull request merged**